core/src/validator.rs

```diff
@@ -2061,13 +2063,65 @@ fn get_stake_percent_in_gossip(bank: &Bank, cluster_info: &ClusterInfo, log: boo
     online_stake_percentage as u64
 }
 
-// Cleanup anything that looks like an accounts append-vec
-fn cleanup_accounts_path(account_path: &std::path::Path) {
-    if let Err(e) = std::fs::remove_dir_all(account_path) {
-        warn!(
-            "encountered error removing accounts path: {:?}: {}",
-            account_path, e
-        );
+/*  Rename to-be-deleted path to *_delete, and add it into the del_paths vector
+ *  If the _delete path exists (could happen if the previous process was killed),
+ *  then append {count} to the path, and delete them together.  This avoids file
+ *  leaking and accumulation even the asyc deleting tasks were not successfully done
+ */
+fn move_path_for_async_removal(del_paths: &mut Vec<PathBuf>, path: &std::path::Path) {
+    let mut del_path_str = OsString::from(path);
+    del_path_str.push("_deleted");
+    let mut loop_count: i32 = 0;
+    loop {
+        let mut with_appendix = del_path_str.clone();
+        if loop_count > 0 {
+            with_appendix.push(format!("{loop_count}"));
+        }
+        let path_delete: PathBuf = PathBuf::from(with_appendix);
+
+        if path_delete.exists() {
+            del_paths.push(path_delete);
+            loop_count += 1;
+        } else {
+            std::fs::rename(&path, &path_delete).unwrap();
+            del_paths.push(path_delete);
+            break;
+        }
+    }
+}
+
+/*  Delete directories/files asynchronously to avoid blocking on it.
```

Comment:
probably `///` will be better for these comments. Or maybe /** works? @brooksprumo

Reply:
I can change those lines to ///
It looks it rust has some comment rules.  inner blocks and ouer blocks are different.  
https://doc.rust-lang.org/reference/comments.html